mso/resource_mso_schema_template_contract_service_graph.go

```diff
@@ -1085,229 +392,83 @@ func resourceTemplateContractServiceGraphRead(d *schema.ResourceData, m interfac
+// PostSchemaTemplateContractServiceGraphConfig creates a new service graph configuration for a template contract.
+//
+// Parameters:
+//   - action: The action to perform.
+//   - d: The schema resource data.
+//   - m: The client interface.
+//
+// Returns:
+//   - An error if there was a problem creating the service graph configuration.
+func PostSchemaTemplateContractServiceGraphConfig(action string, d *schema.ResourceData, m interface{}) error {
 	msoClient := m.(*client.Client)
-
-	schemaID := d.Get("schema_id").(string)
-	siteID := d.Get("site_id").(string)
+	schemaId := d.Get("schema_id").(string)
 	TemplateName := d.Get("template_name").(string)
 	contractName := d.Get("contract_name").(string)
+	serviceGraph := d.Get("service_graph_name").(string)
 
-	tempPath := fmt.Sprintf("/templates/%s/contracts/%s/serviceGraphRelationship", TemplateName, contractName)
-	tempConGraph := models.NewTemplateContractServiceGraph("remove", tempPath, nil, nil)
-
-	sitePath := fmt.Sprintf("/sites/%s-%s/contracts/%s/serviceGraphRelationship", siteID, TemplateName, contractName)
-	siteConGraph := models.NewSiteContractServiceGraph("remove", sitePath, nil, nil)
-
-	response, err := msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", schemaID), tempConGraph, siteConGraph)
-
-	// Ignoring Error with code 141: Resource Not Found when deleting
-	if err != nil && !(response.Exists("code") && response.S("code").String() == "141") {
-		return err
-	}
-
-	d.SetId("")
-	log.Printf("[DEBUG] Completed Delete Template Contract Service Graph")
-	return nil
-}
+	serviceGraphRef := make(map[string]interface{})
+	serviceGraphRef["serviceGraphName"] = serviceGraph
 
-func getTemplateServiceGraph(cont *container.Container, templateName, graphName string) (*container.Container, int, error) {
-	templateCount, err := cont.ArrayCount("templates")
-	if err != nil {
-		return nil, -1, fmt.Errorf("No Template found")
+	if graphSchema, ok := d.GetOk("service_graph_schema_id"); ok {
+		serviceGraphRef["schemaId"] = graphSchema.(string)
+	} else {
+		serviceGraphRef["schemaId"] = schemaId
 	}
 
-	for i := 0; i < templateCount; i++ {
-		templateCont, err := cont.ArrayElement(i, "templates")
-		if err != nil {
-			return nil, -1, fmt.Errorf("Unable to get template element")
-		}
-
-		apiTemplate := models.StripQuotes(templateCont.S("name").String())
-
-		if apiTemplate == templateName {
-			log.Printf("[DEBUG] Template found")
-
-			sgCount, err := templateCont.ArrayCount("serviceGraphs")
-
-			if err != nil {
-				return nil, -1, fmt.Errorf("No Service Graph found")
-			}
-
-			for j := 0; j < sgCount; j++ {
-				sgCont, err := templateCont.ArrayElement(j, "serviceGraphs")
-
-				if err != nil {
-					return nil, -1, fmt.Errorf("Unable to get service graph element")
-				}
-
-				apiSgName := models.StripQuotes(sgCont.S("name").String())
-
-				if apiSgName == graphName {
-					return sgCont, j, nil
-				}
-			}
-
-		}
+	if graphTemp, ok := d.GetOk("service_graph_template_name"); ok {
```

Comment:
should be keep the naming aligned with the schema definition?
```suggestion
	if serviceGraphTemplateName, ok := d.GetOk("service_graph_template_name"); ok {
```